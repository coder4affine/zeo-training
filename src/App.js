import React, { Component } from "react";
import PropTypes from "prop-types";
import TodoForm from "./todoForm";

export default class App extends Component {
  static propTypes = {};

  state = {
    value: "",
    todoList: [],
    filteredList: [],
    currentState: "all"
  };

  componentDidMount() {
    document.addEventListener("copy", this.onCopy);
  }

  componentWillUnmount() {
    document.removeEventListener("copy", this.onCopy);
  }

  onCopy = () => {
    alert("copied");
  };

  onChange = e => {
    this.setState({ value: e.target.value });
  };

  submit = e => {
    console.log(e);
    e.preventDefault();
    const { value, todoList, currentState } = this.state;
    this.setState(
      {
        todoList: [
          ...todoList,
          {
            id: new Date().getUTCMilliseconds(),
            todoText: value,
            status: "pending"
          }
        ],
        value: ""
      },
      () => {
        this.filterData(currentState);
      }
    );
  };

  changeStatus = data => {
    const { todoList, currentState } = this.state;

    const index = todoList.findIndex(x => x.id === data.id);
    const updatedData = [
      ...todoList.slice(0, index),
      { ...data, status: data.status === "pending" ? "done" : "pending" },
      ...todoList.slice(index + 1)
    ];
    this.setState({ todoList: updatedData }, () => {
      this.filterData(currentState);
    });
  };

  deleteRecord = data => {
    const { todoList, currentState } = this.state;

    const index = todoList.findIndex(x => x.id === data.id);
    const updatedData = [
      ...todoList.slice(0, index),
      ...todoList.slice(index + 1)
    ];
    this.setState({ todoList: updatedData }, () => {
      this.filterData(currentState);
    });
  };

  filterData = type => {
    const filteredList = this.getFilterData(type);
    this.setState({ filteredList, currentState: type });
  };

  getFilterData = type => {
    const { todoList } = this.state;
    let filteredList = [];
    switch (type) {
      case "pending":
        filteredList = todoList.filter(x => x.status === "pending");
        break;
      case "done":
        filteredList = todoList.filter(x => x.status === "done");
        break;

      default:
        filteredList = todoList;
        break;
    }
    return filteredList;
  };

  render() {
    const { value, filteredList } = this.state;
    return (
      <div>
        <h3> TODO Application</h3>
        {/* form  */}
        <TodoForm
          onSubmit={this.submit}
          value={value}
          onChange={this.onChange}
          label="Todo"
        />
        {/* data */}
        {filteredList.map((x, index) => (
          <div key={index}>
            <span onClick={() => this.changeStatus(x)}>
              <span>{x.todoText}</span>
              <span style={{ color: "green" }}> {x.status}</span>
            </span>
            <span style={{ color: "red" }} onClick={() => this.deleteRecord(x)}>
              Delete
            </span>
          </div>
        ))}
        {/* filtering */}
        <button onClick={() => this.filterData("all")}>All Task</button>
        <button onClick={() => this.filterData("done")}>Completed Task</button>
        <button onClick={() => this.filterData("pending")}>Pending Task</button>
      </div>
    );
  }
}
