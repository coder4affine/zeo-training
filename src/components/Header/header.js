import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import Loading from "../Loading/loading";

const header = ({ loading }) => {
  return (
    <nav>
      <Link to="/" activeClassName="active">
        Home
      </Link>
      {"|"}
      <Link to="/courses" activeClassName="active">
        Courses
      </Link>
      {"|"}
      <Link to="/about" activeClassName="about">
        About
      </Link>
      {loading && <Loading />}
    </nav>
  );
};

header.PropTypes = {
  loading: PropTypes.bool
};

header.defaultProps = {
  loading: false
};

export default header;
