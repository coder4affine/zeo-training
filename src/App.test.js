import React from "react";
import ReactDOM from "react-dom";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Header from "./components/Header/header";
import App from "./App";
configure({ adapter: new Adapter() });

describe("snapshot", () => {
  it("snapshot", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});

it("renders without crashing", () => {
  const wrapper = shallow(<App />);
  expect(wrapper.contains(<h1>Zeolearn APP</h1>)).toEqual(true);
  expect(wrapper.contains(<Header />)).toEqual(true);
});
