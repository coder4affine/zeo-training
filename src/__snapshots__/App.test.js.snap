// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <BrowserRouter>
        <React.Fragment>
          <h1>
            Zeolearn APP
          </h1>
          <header
            loading={false}
          />
          <Context.Provider
            value={
              Object {
                "changeLanguage": [Function],
                "language": "en",
              }
            }
          >
            <route />
          </Context.Provider>
        </React.Fragment>
      </BrowserRouter>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <React.Fragment>
          <h1>
            Zeolearn APP
          </h1>
          <header
            loading={false}
          />
          <Context.Provider
            value={
              Object {
                "changeLanguage": [Function],
                "language": "en",
              }
            }
          >
            <route />
          </Context.Provider>
        </React.Fragment>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <h1>
              Zeolearn APP
            </h1>,
            <header
              loading={false}
            />,
            <Context.Provider
              value={
                Object {
                  "changeLanguage": [Function],
                  "language": "en",
                }
              }
            >
              <route />
            </Context.Provider>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Zeolearn APP",
            },
            "ref": null,
            "rendered": "Zeolearn APP",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "loading": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <route />,
              "value": Object {
                "changeLanguage": [Function],
                "language": "en",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.provider),
              "_context": Object {
                "$$typeof": Symbol(react.context),
                "Consumer": Object {
                  "$$typeof": Symbol(react.context),
                  "_calculateChangedBits": null,
                  "_context": [Circular],
                },
                "Provider": [Circular],
                "_calculateChangedBits": null,
                "_currentRenderer": null,
                "_currentRenderer2": null,
                "_currentValue": undefined,
                "_currentValue2": undefined,
                "_threadCount": 0,
              },
            },
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BrowserRouter>
          <React.Fragment>
            <h1>
              Zeolearn APP
            </h1>
            <header
              loading={false}
            />
            <Context.Provider
              value={
                Object {
                  "changeLanguage": [Function],
                  "language": "en",
                }
              }
            >
              <route />
            </Context.Provider>
          </React.Fragment>
        </BrowserRouter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <React.Fragment>
            <h1>
              Zeolearn APP
            </h1>
            <header
              loading={false}
            />
            <Context.Provider
              value={
                Object {
                  "changeLanguage": [Function],
                  "language": "en",
                }
              }
            >
              <route />
            </Context.Provider>
          </React.Fragment>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h1>
                Zeolearn APP
              </h1>,
              <header
                loading={false}
              />,
              <Context.Provider
                value={
                  Object {
                    "changeLanguage": [Function],
                    "language": "en",
                  }
                }
              >
                <route />
              </Context.Provider>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Zeolearn APP",
              },
              "ref": null,
              "rendered": "Zeolearn APP",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "loading": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <route />,
                "value": Object {
                  "changeLanguage": [Function],
                  "language": "en",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.provider),
                "_context": Object {
                  "$$typeof": Symbol(react.context),
                  "Consumer": Object {
                    "$$typeof": Symbol(react.context),
                    "_calculateChangedBits": null,
                    "_context": [Circular],
                  },
                  "Provider": [Circular],
                  "_calculateChangedBits": null,
                  "_currentRenderer": null,
                  "_currentRenderer2": null,
                  "_currentValue": undefined,
                  "_currentValue2": undefined,
                  "_threadCount": 0,
                },
              },
            },
          ],
          "type": Symbol(react.fragment),
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
